### ISSUE TITLE
[Bug]  submodule  _protocol is not available on GitHub mirror

### ISSUE THREAD

**Issue by andrewcrook** [HiPhish/fennel-repl.nvim#1]
The submodule  _protocol is not available on GitHub mirror
Anyone using a package manager to install from GitHub will find the package fails to install. 

Even if for example with Lazy
I use a custom url with the gitlab repo it will fail with the submodule
I don't think submodules are supported
Couldn’t a copy of _protocol just be included ?
or some sort of installation script to install the submodule into the directory separately on install and updates ?

**Comment by HiPhish** [comment]
That sounds to me like a feature problem with your package manager. Do note that this plugin is nowhere near ready for use by the public. It suffers the traditional curse of everything Lisp: I got it done 80% of the way and then left it unfinished because I do not use it in my day-to-day programming. The protocol itself is not stable either, and the last commit has been nearly half a year ago.

> Couldn’t a copy of _protocol just be included ?

I would rather avoid this because it means I have to keep code I have not written in the repo. Git submodules are an elegant solution, even if not the best one.

> or some sort of installation script to install the submodule into the directory separately on install and updates ?

That would be a possibility, a shell script that manually executes `git submodule ...`. You would then have to set up your package manager to call that script automatically after each update. What would such a script need to look like from the outside?

**Comment by andrewcrook** [comment]
@HiPhish 

It seems Lazy does actually support submodules so its not that.
Git doesn’t work with it 
I think this is more to do with the github repo  itself  the sub module link on GitHub itself doesn’t lead anywhere its not found.  
I don't know git expects the submodule to be on the same platform as well? .. never tried it.
I think the submodule needs to be recreated and , possibly, a mirror for the other repo on GitHub ?

**Comment by HiPhish** [comment]
> I don't know git expects the submodule to be on the same platform as well?

No, Git itself does not care whether the submodule is on a different server than the main repo.  Maybe your Git configuration is messed up.  You can try these steps yourself and see if it works on your machine:

```sh
git clone https://github.com/HiPhish/fennel-repl.nvim/
cd fennel-repl.nvim
git submodule update --init _protocol
git submodule update --checkout _protocol
```

If this works then we know it is an issue with your package manager.

> I think the submodule needs to be recreated and , possibly, a mirror for the other repo on GitHub ?

I don't think you can have a different submodule URL depending on where the repo is hosted.  This repo is a straight-up clone of the GitLab repo and GitLab automatically pushes any commits it receives here.  What you are proposing is have to separate repos that I have to manually keep in sync.

**Comment by andrewcrook** [comment]
Thanks, I’ll check it later but I now got a feeling it is because gitlab doesn’t allow cloning submodules without an account.

**Comment by HiPhish** [comment]
@andrewcrook Any updates?
